cli/cmd/uninstall.go

```diff
@@ -38,9 +39,17 @@ var uninstallCmd = &cobra.Command{
 
 		ns, err := resources.GetOdigosNamespace(client, ctx)
 		if err != nil {
-			ns = consts.DefaultNamespace
-		}
+			errString := err.Error()
 
+			if errString[len(errString) - 1] == '0' {
```

Comment:
Can you please describe what this code checks in practice?

Reply:
This code checks for the condition if the namespace is not found

Reply:
Is this because you count on the error message to contain the specific text: `"expected to get 1 namespace got 0"`?

I have a few concerns regarding this practice:
1. What if someone in the future decides to change this error text? then the feature will break and it will be tough for anyone to figure it out.
2. There is also the case that [some general `err` is returned](https://github.com/keyval-dev/odigos/blob/bdf9a967b8960b7e55e06e4d5daa4074290a6ffc/cli/cmd/resources/namespace.go#L29). For this error, we cannot know for sure if it ends with the character `0` or not. Thus, making the current logic less robust and inviting bugs.
3. The code line `if errString[len(errString) - 1] == '0' {` is not trivial to understand for future maintainers and readers of this repo. It does not describe what is checked here and why it is expected to work.

You can consider using [`errors.Is`](https://pkg.go.dev/errors#Is) to test if an error is of specific type. There are numerous blog posts in google and even some examples in this repo on how to use it.

Reply:
Your concern is absolutely right,
I am thinking about something like:
```go

ns, err := resources.GetOdigosNamespace(client, ctx)
if err != nil {
  var ErrOdigosNotInstalled = errors.New("expected to get 1 namespace got 0")
  if errors.Is(err, ErrOdigosNotInstalled) {
    fmt.Println("\033[31mERROR\033[0m", Odigos is not currently installed in the cluster, so there is nothing to uninstall )
    } else {
    fmt.Println("\033[31mERROR\033[0m", err)
    }
  }
 os.Exit(1)
}
```

Reply:
Correct me if i am wrong